src/store/vuex.ts

```diff
@@ -0,0 +1,190 @@
+import { InjectionKey } from "vue";
+import {
+  Store as BaseStore,
+  createStore as baseCreateStore,
+  useStore as baseUseStore,
+  DispatchOptions,
+  CommitOptions,
+  ModuleTree,
+  Plugin,
+  StoreOptions as OriginalStoreOptions,
+  GetterTree as OriginalGetterTree,
+  ActionTree as OriginalActionTree,
+  MutationTree as OriginalMutationTree,
+} from "vuex";
+
+export type PayloadFunction = (payload?: any) => any;
+
+export type GettersBase = Record<string, any>;
+export type ActionsBase = Record<string, PayloadFunction>;
+export type MutationsBase = Record<string, any>;
+
+export class Store<
+  S,
+  G extends GettersBase,
+  A extends ActionsBase,
+  M extends MutationsBase
+> extends BaseStore<S> {
+  constructor(options: OriginalStoreOptions<S>) {
+    super(options);
+  }
+
+  readonly getters!: G;
+
+  // 既に型がつけられているものを上書きすることになるので、TS2564を吐く、それの回避
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  dispatch: Dispatch<A>;
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  commit: Commit<M>;
+}
+
+export function createStore<
+  S,
+  G extends GettersBase,
+  A extends ActionsBase,
+  M extends MutationsBase
+>(options: StoreOptions<S, G, A, M>): Store<S, G, A, M> {
+  // optionsをOriginalStoreOptions<S>で型キャストしないとTS2589を吐く
+  return baseCreateStore<S>(options as OriginalStoreOptions<S>) as Store<
+    S,
+    G,
+    A,
+    M
+  >;
+}
+
+export function useStore<
+  S,
+  G extends GettersBase,
+  A extends ActionsBase,
+  M extends MutationsBase
+>(injectKey?: InjectionKey<Store<S, G, A, M>> | string): Store<S, G, A, M> {
+  return baseUseStore<S>(injectKey) as Store<S, G, A, M>;
+}
+
+export interface Dispatch<A extends ActionsBase> {
+  <T extends keyof A>(
+    type: T,
+    payload: Parameters<A[T]>[0],
+    options?: DispatchOptions
+  ): Promise<ReturnType<A[T]>>;
+  <T extends keyof A>(
+    payloadWithType: { type: T } & (Parameters<A[T]>[0] extends Record<
+      string,
+      any
+    >
+      ? Parameters<A[T]>[0]
+      : // eslint-disable-next-line @typescript-eslint/ban-types
+        {}),
+    options?: DispatchOptions
+  ): Promise<ReturnType<A[T]>>;
+}
+
+export interface Commit<M extends MutationsBase> {
+  <T extends keyof M>(type: T, payload: M[T], options?: CommitOptions): void;
```

Comment:
#269 では、Mutationにおいてpayloadが必要でなければ、`undefined`を指定している([例](https://github.com/Hiroshiba/voicevox/pull/269/files#diff-035c66227ff9c0b4b1c690e921c7bb816eb92dbaca565f81ab7ba906bed613a8R36-R37))のですが、現状の設計では、`commit`を呼び出すときに`undefined`を入力しなければなりません。
![image](https://user-images.githubusercontent.com/34832037/134932476-f263408c-d079-4f67-81c3-9cc290de8a35.png)

これを回避するには、以下のようにすればよいのですが、そうするとpayloadに`undefined`を指定しているもの以外もoptionalになります。
```suggestion
  <T extends keyof M>(type: T, payload?: M[T], options?: CommitOptions): void;
```
`undefined`を引数に持たせなければならないのは若干気持ち悪いですが、optionalにすると型の安全性が確保できないので、現状はこうしています...

なお、Dispatchも同様な形になっています。